addons/pkg/util/clusterbootstrapclone/clusterbootstrapclone.go

```diff
@@ -171,15 +209,203 @@ func addMissingFields(copyFrom, destination map[string]interface{}) error {
 				// If we do not have the following check the final processed ClusterBootstrap will have {"a": "b", "foo": "bar"}
 				// which is not expected
 				keyInFrom != "inline" {
-				if err := addMissingFields(valueInFrom.(map[string]interface{}), valueInTarget.(map[string]interface{})); err != nil {
+				if err := addMissingFields(valueInFrom.(map[string]interface{}), valueInTarget.(map[string]interface{}), keysToSkip); err != nil {
 					return err
 				}
+			} else if valueInFrom != nil && reflect.TypeOf(valueInFrom).Kind() == reflect.Slice &&
+				valueInTarget != nil && reflect.TypeOf(valueInTarget).Kind() == reflect.Slice &&
+				keyInFrom == "additionalPackages" {
```

Comment:
Question: Would it make this logic simpler if we we did not have to merge additionalPackages? We would only handle copying additionalPackages if key did not exist in destination.

Note: Don't need to make the change here, lets just log issue if we need to simplify this.